src/services/OswServiceTaskCore.cpp

```diff
@@ -0,0 +1,112 @@
+#include <time.h>
```

Comment:
Could your rename this task to `Steps`? As the task currently _only_ handles the step counting with the history and the name `core` is ambiguous.

Reply:
Yeah my end goal is to actually perform all core functions here like:

```
  hal->handleWakeupFromLightSleep();
  hal->checkButtons();
  hal->fetchRtcTime();
 ```
What are your toughts ?

Reply:
>  hal->handleWakeupFromLightSleep();

YES

>  hal->fetchRtcTime();

Yes

>  hal->checkButtons();

Maybe? This is user interaction and therefore time critical - there is grantee that we won't slow down the services further down the road. You know what? I'll open an issue to rewrite the manager to respect time-critical services.

Reply:
But for now, it does not has those - right? Therefore for now, it is only the step service. Also, why would you try to merge all those functions into one service task - wouldn't many small be better? Stack size... Yes... Okay, point made for now...